aea/helpers/dialogue/base.py

```diff
@@ -400,6 +404,24 @@ def update(self, message: Message) -> bool:
                 self._outgoing_messages.extend([message])
         return is_extendable
 
+    def reply(
+        self, target_message: Message, performative, **kwargs
+    ) -> Optional[Message]:
+        reply = self._message_class(
+            dialogue_reference=self.dialogue_label.dialogue_reference,
+            message_id=self.last_message.message_id + 1,
+            target=target_message.message_id,
+            performative=performative,
+            **kwargs,
+        )
+        reply.counterparty = self.dialogue_label.dialogue_opponent_addr
+        result = self.update(reply)
+
+        if result:
+            return reply
+        else:
+            return None
```

Comment:
I think we should throw an error; because the dialogue not being constructed means we have specified something incorrectly in the kwargs; which should be thrown rather than handled.

Reply:
Agree.

Reply:
Well, after changing the errors I realised its going to change the interface (the return values of methods). So need to hold this off for the next release if we want to be backwards compatible for this release. I make a list of all breaking changes for the next release to not forget anything.

Reply:
reply doesn;t exist in 0.5.0 so you can do it here: raise if reply is None

Reply:
Done